noir/noir-repo/examples/recursion/recurse_leaf/src/main.nr

```diff
@@ -0,0 +1,20 @@
+use dep::std;
+
+#[recursive]
+fn main(
+    verification_key: [Field; 114],
+    public_inputs: pub [Field; 3],
+    key_hash: Field,
+    proof: [Field; 93],
+    num: u64
+) -> pub u64 {
+    // verify sum so far was computed correctly
+    std::verify_proof(
+        verification_key.as_slice(),
+        proof.as_slice(),
+        public_inputs.as_slice(),
+        key_hash
+    );
+    // Take output of previous proof and add another number to it.
+    public_inputs[2] as u64 + num
```

Comment:
Although the SumLib was trivial, I thought it was helpful to test a call inside nested recursive verification, from innermost proof and in the leaf proof. (Esp with Noir folding coming in?)

Reply:
This shouldn't make any difference tbh as we'll just inline the function call anyway so the ACIR will be unchanged.